### PR TITLE
Fixed #34861 -- Fixed crash when adding GeneratedFields with some expressions.

### PR DESCRIPTION
**Pull request by pauloxnet**
Ref https://code.djangoproject.com/ticket/34861

### CODE REVIEW

File: tests/model_fields/test_generatedfield.py [hunk 1]
```diff
@@ -16,6 +21,31 @@
...
+        class Response(Model):
+            data = JSONField(default=dict)
+            status = GeneratedField(
+                db_persist=True,
+                expression=KT("data__status"),
+            )
+
+        with connection.schema_editor(collect_sql=True) as editor:
+            editor.create_model(Response)
+            self.assertNotIn("None", "\n".join(editor.collected_sql))
```
Comment (felixxm) [inline tests/model_fields/test_generatedfield.py:46]: I'd move it to the `tests/schema/tests.py`.

File: tests/model_fields/test_generatedfield.py [hunk 2]
```diff
@@ -16,6 +21,31 @@
 )
 
 
+@isolate_apps("model_fields")
+class GeneratedFieldTransactionTests(TransactionTestCase):
+    available_apps = ["model_fields"]
+
+    def test_generated_sql(self):
```
[the remark this answers is not in the document]
Comment (pauloxnet) [reply tests/model_fields/test_generatedfield.py:28]: I thought all Django database backends supported JSONField or GeneratedField. Am I wrong?

File: tests/model_fields/test_generatedfield.py (hunk 1 above)
Comment (pauloxnet) [reply tests/model_fields/test_generatedfield.py:46]: Ok, I'll do that.

File: tests/model_fields/test_generatedfield.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (felixxm) [reply tests/model_fields/test_generatedfield.py:28]: > I thought all Django database backends supported JSONField or GeneratedField. Am I wrong?

It doesn't matter, we have to respect our own feature flags. Also, 3rd-party database backends are using our test suite. Last but not least, generated fields are not supported by all databases, e.g. SQLite < 3.31.

[the remark this answers is not in the document]
Comment (pauloxnet) [reply tests/model_fields/test_generatedfield.py:28]: To skip the test when `JSONfield` is not supported I'll add:
```
@skipUnlessDBFeature("supports_json_field")
```

For `GeneratedField` I can't find a similar feature flag do you think this decorator is correct?

```
@skipUnlessAnyDBFeature(
    "supports_stored_generated_columns", "supports_virtual_generated_columns"
)
```

[the remark this answers is not in the document]
Comment (felixxm) [reply tests/model_fields/test_generatedfield.py]: You used `db_persist=True` so `supports_stored_generated_columns` should be used.

File: tests/schema/tests.py [hunk 3]
```diff
@@ -5298,3 +5303,29 @@ def test_ci_cs_db_collation(self):
             if connection.vendor == "postgresql":
                 with connection.cursor() as cursor:
                     cursor.execute("DROP COLLATION IF EXISTS case_insensitive")
+
+
+@isolate_apps("schema")
+@skipUnlessAnyDBFeature("supports_json_field", "supports_stored_generated_columns")
```
Comment (felixxm) [inline tests/schema/tests.py:5309]: Both must be supported
```suggestion
@skipUnlessDBFeature("supports_json_field", "supports_stored_generated_columns")
```

File: tests/model_fields/test_generatedfield.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (pauloxnet) [reply tests/model_fields/test_generatedfield.py]: > You used `db_persist=True` so `supports_stored_generated_columns` should be used.

Thanks.

The PR should be ready now.

File: tests/schema/tests.py [hunk 4]
```diff
@@ -5298,3 +5303,29 @@ def test_ci_cs_db_collation(self):
...
+    available_apps = ["schema"]
+
+    def test_generated_sql(self):
+        """
+        Regression test for #34861
+
+        GeneratedField.generated_sql shouldn't generate broken SQL code.
+        """
+        from django.db.models import GeneratedField, JSONField
+        from django.db.models.fields.json import KT
```
Comment (felixxm) [inline tests/schema/tests.py:5320]: Inline imports are not necessary.

File: tests/schema/tests.py (hunk 3 above)
Comment (pauloxnet) [reply tests/schema/tests.py]: Fixed

File: tests/schema/tests.py (hunk 4 above)
Comment (pauloxnet) [reply tests/schema/tests.py:5315]: Your message is a little too vague or polite. :)

Are you just commenting that inline imports aren't necessary or are you asking me to move these imports up for you to consider the PR mergeable?

Comment (felixxm) [reply tests/schema/tests.py:5315]: :smile: Yes, please move them to the top :100:

Comment (pauloxnet) [reply tests/schema/tests.py]: Done :)

**Review by felixxm — APPROVED** [review]
@pauloxnet Thanks :+1: 

I moved a test method to the main class.

**Comment by pauloxnet** [comment]
> @pauloxnet Thanks 👍

Thanks for your precious review 🙏🏻

**Pull request merged**